### ISSUE TITLE
WPS434 on tuple conversion

### ISSUE THREAD

**Issue by vnmabus** [wemake-services/wemake-python-styleguide#1807]
# Bug report

## What's wrong

A `WPS434: Found reassigning variable to itself` violation is raised for the following code:
```python
my_var = (my_var,)
```

## How it should be

As the code replaces the original object with a tuple containing it, the line is not a simple reasignation. Thus, the violation WPS434 should not be raised.

**Comment by orsinium** [comment]
FYI, MyPy won't be happy with this code as well unless you have [allow_redefinition](https://mypy.readthedocs.io/en/stable/config_file.html#confval-allow_redefinition) enabled. SO, probably it's worth making a new variable here.

**Comment by vnmabus** [comment]
It seems that redefinitions that narrow the original type are ok in Mypy. In my case this code is being used for a variable that can be either a tuple or an individual element, converting it into a one element tuple in that case. Mypy does not complain even with strict flags enabled, and `reveal_type` reflects the type narrowing correctly.

**Comment by sobolevn** [comment]
I would leave this as `wontfix`, because this is a very special case.
I would recommend to:
- Use `noqa` for this line
- Use other variable name

**Comment by vnmabus** [comment]
The same problem happens if you do `my_var = (my_var, my_var)`. I know you said you will not fix it, but just in case.
